GOSSIP.md

```diff
@@ -2,30 +2,31 @@
 
 As explained in the [Sync](SYNC.md) document, any blockchain implementation
 needs communication with peers in order to be truly useful. Apart from the lower
-level details described in that document, there is the additional dimension of
+level details described in that document, there is the added dimension of
 how peers are selected for communication, how connections to those peers are
 maintained over time. This is the purpose of gossiping functionality.
 
 ## Initial Configuration
 
-The initial set of peers is defined from configuration when starting the node.
+The initial set of peers is defined by configuration when starting the node.
 This is a predefined set of peers that are automatically connected to upon
 startup.
 
 ## Peers
 
-A peer is identified by a URI consisting of the protocol `aenode://`, the public
-key, an `@` character, the hostname or IP number, a `:` character and the Noise
-port number the node is using. Example:
+A peer is identified by a URI consisting of the protocol 'aenode://', the public
+key, an '@' character, the hostname or IP number, a ':' character and the Noise
+port number the node is using. If the address contains a hostname, it will be
+resolved to an IP. Example:
 
-```
+'''
 aenode://pp$HryRGHJ7Ct3trkktVyVBgfhHL1J4EYSD9cScuMZDV61eSHrCZ@mynode.example.com:3015
-```
+'''
 
-A node is uniquely identified by the triplet of public key, hostname and port.
-This means one hostname can have several instanced of Epoch started at the same
+A node is uniquely identified by the triplet of public key, IP and port.
```

Comment:
This is actually wrong, or at least not according to the implementation, a node is uniquely identified by the public key only!

Reply:
But practically it is bounded by the IP/port as well, right? You can't have two nodes with different keys connected through the same socket, no? 🙂

Reply:
Maybe it should be reworded, the pubkey is the only "identity" of a node, but there is a single IP/port for a not to. There couldn't be multiple IP/port with the same pubkey or multiple pubkey with the same IP/port. Is that correct?

Reply:
If we have a node with a dynamic IP address then we might end up with multiple `(IPADDR, PORT)` pairs with the same pubkey, although that should be a transient state. (Or maybe if a node has multiple interfaces and listens on them too?)

Multiple pubkeys with the same `(IPADDR, PORT)` pair should not be possible.

Reply:
So we should remove any known peer with the same pubkey but different IP/port before adding a new peer.